index.bs

```diff
@@ -657,132 +882,160 @@ dictionary FilePropertyBag : BlobPropertyBag {
+
 {{File}} objects are [=serializable objects=]. Their [=serialization steps=],
-given |value| and |serialized|, are:
+given |value|, |serialized| and |forStorage|, are:
 
-1. Set |serialized|.\[[SnapshotState]] to |value|'s [=snapshot state=].
+1. Invoke the [=blob serialization steps=] given |value|, |serialized| and |forStorage|.
+1. Set |serialized|.\[[Name]] to |value|.[=[[name]]=].
+1. Set |serialized|.\[[LastModified]] to |value|.[=[[lastModified]]=].
 
-2. Set |serialized|.\[[ByteSequence]] to |value|'s underlying byte sequence.
+Their [=deserialization steps=], given |value| and |serialized|, are:
 
-3. Set |serialized|.\[[Name]] to the value of |value|'s {{File/name}} attribute.
+1. Invoke the [=blob deserialization steps=] given |value| and |serialized|.
+1. Set |value|.[=[[name]]=] to |serialized|.\[[Name]].
+1. Set |value|.[=[[lastModified]]=] to |serialized|.\[[LastModified]].
 
-4. Set |serialized|.\[[LastModified]] to the value of |value|'s {{File/lastModified}} attribute.
+## Concepts ## {#file-concepts}
 
-Their [=deserialization steps=], given |value| and |serialized|, are:
+The [=snapshot state=] for a file backed blob contains
+a <dfn for="file blob snapshot state"><code>"file"</code></dfn> member (a reference to a native file on disk),
+and a <dfn for="file blob snapshot state"><code>"last modified"</code></dfn> member (a number).
+
+<div algorithm="create a file">
+To <dfn export>create a file backed {{File}} object</dfn> for a given |native file|,
+run these steps:
+
+1. Let |snapshot state| be an empty [=map=].
+1. Set |snapshot state|[<code><a for="file blob snapshot state">"file"</a></code>] to |native file|.
+1. Let |last modified| be the last time |native file| was modified,
+  as the number of milliseconds since the [=Unix Epoch=].
+  If this can't be determined, set |last modified| to the current date and time
+    represented as the number of milliseconds since the [=Unix Epoch=].
+1. Set |snapshot state|[<code><a for="file blob snapshot state">"last modified"</a></code>] to |last modified|.
+
+1. Let |name| be the file name of |native file|, converted to a string in a
+  user agent defined manner.
+1. Let |content type| be the mime type of |native file| (as a lowercase ASCII string),
+  derived from |name| in a user agent defined manner, or an empty string if no type
+  could be determined, taking into account the following <dfn export>file type guidelines</dfn>:
+
+  * User agents must return the {{Blob/type}} as an ASCII-encoded string in lower case,
+    such that when it is converted to a corresponding byte sequence,
+    it is a <a>parsable MIME type</a>,
+    or the empty string &ndash; 0 bytes &ndash; if the type cannot be determined.
+  * When the file is of type <code>text/plain</code>
+    user agents must NOT append a charset parameter to the <i>dictionary of parameters</i> portion of the media type [[!MIMESNIFF]].
+  * User agents must not attempt heuristic determination of encoding,
```

Comment:
Since encoding is not specified for `text/plain` (per previous), this can't apply in that case. Should this be scoped to other text/* types?